web/packages/teleport/src/Bots/Add/GitHubActions/useGitHubFlow.tsx

```diff
@@ -0,0 +1,275 @@
+/**
+ * Teleport
+ * Copyright (C) 2023  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { useState, useContext } from 'react';
+import { Option } from 'shared/components/Select';
+
+import useAttempt, { Attempt } from 'shared/hooks/useAttemptNext';
+
+import { ResourceLabel } from 'teleport/services/agents';
+import {
+  createBot as serviceCreateBot,
+  createBotToken,
+} from 'teleport/services/bot';
+import {
+  CreateBotRequest,
+  GitHubRepoRule,
+  RefType,
+} from 'teleport/services/bot/types';
+
+import useTeleport from 'teleport/useTeleport';
+
+export const GITHUB_HOST = 'github.com';
+const GITHUB_ACTIONS_LABEL_KEY = 'teleport.internal/ui-flow';
+const GITHUB_ACTIONS_LABEL_VAL = 'github-actions-ssh';
+
+type GitHubFlowContext = {
+  attempt: Attempt;
+  createBotRequest: CreateBotRequest;
+  setCreateBotRequest: React.Dispatch<React.SetStateAction<CreateBotRequest>>;
+  repoRules: Rule[];
+  setRepoRules: React.Dispatch<React.SetStateAction<Rule[]>>;
+  addEmptyRepoRule: () => void;
+  tokenName: string;
+  createBot: () => Promise<boolean>;
+  resetAttempt: () => void;
+};
+
+const gitHubFlowContext = React.createContext<GitHubFlowContext>(null);
+
+export const initialBotState = {
+  labels: [{ name: '*', value: '*' }],
+  login: '',
+  botName: '',
+  roles: [],
+};
+
+export function GitHubFlowProvider({
+  children,
+  bot = initialBotState,
+}: React.PropsWithChildren<{ bot?: CreateBotRequest }>) {
+  const { resourceService } = useTeleport();
+  const { attempt, run } = useAttempt();
+  const [createBotRequest, setCreateBotRequest] =
+    useState<CreateBotRequest>(bot);
+  const [repoRules, setRepoRules] = useState<Rule[]>([defaultRule]);
+  const [tokenName, setTokenName] = useState('');
+
+  function addEmptyRepoRule() {
+    setRepoRules([...repoRules, defaultRule]);
+  }
+
+  function resetAttempt() {
+    if (attempt.status !== 'processing') {
+      attempt.status = '';
```

Comment:
did you mean, i dont think the way it is done^ will trigger a re-render
```suggestion
     setAttempt({status: ''})
```

Reply:
Yes, thank you!